### PR TITLE
[Improve][Connector-V2] Fake supports column configuration

### LINKED ISSUE #7492: [Feature][Connector-V2] Fake supports column configuration
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Connector-fake supports column configuration
eg：
```
{
    name = book_intro_1
    type = float_vector
    columnScale =4
    comment = "vector"
}
Control the vector dim through columnScale configuration, with the highest priority
```

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by corgy-w**
### Purpose of this pull request
issue : #7492 

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/TableSchema.java [hunk 1]
```diff
@@ -57,6 +57,10 @@ public SeaTunnelRowType toPhysicalRowDataType() {
         return new SeaTunnelRowType(fields, fieldTypes);
     }
 
+    public List<Column> getPhysicalColumns() {
+        return columns.stream().filter(Column::isPhysical).collect(Collectors.toList());
+    }
```
Comment (Hisoka-X) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/TableSchema.java]: All columns are  physical at now. So add new method will let devs confused.

**Review by Hisoka-X — DISMISSED** [review]
LGTM if ci passes. Thanks @corgy-w !

**Comment by corgy-w** [comment]
@Hisoka-X @wuchunfu Also fix dead link,  merge as soon as possible if ci passes

**Comment by hailin0** [comment]
> @Hisoka-X @wuchunfu Also fix dead link, merge as soon as possible if ci passes

@corgy-w 
Can you submit a separate PR to fix this dead link?

**Comment by corgy-w** [comment]
> > @Hisoka-X @wuchunfu Also fix dead link, merge as soon as possible if ci passes
> 
> @corgy-w Can you submit a separate PR to fix this dead link?

get

**Review by Hisoka-X — DISMISSED** [review]
LGTM. Thanks @corgy-w

**Comment by wuchunfu** [comment]
@corgy-w Please resolve the conflict, thanks

**Review by Hisoka-X — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]
LGTM

**Pull request merged**